zapcore/sampler.go

```diff
@@ -62,33 +102,42 @@ func fnv32a(s string) uint32 {
 	return hash
 }
 
-func (c *counter) IncCheckReset(t time.Time, tick time.Duration) uint64 {
+func (c *counter) IncCheckReset(t time.Time, tick time.Duration) (currentVal, preResetVal uint64) {
 	tn := t.UnixNano()
 	resetAfter := c.resetAt.Load()
 	if resetAfter > tn {
-		return c.counter.Inc()
+		return c.counter.Inc(), 0
 	}
 
-	c.counter.Store(1)
+	preResetVal = c.counter.Load()
+	currentVal = 1
+	c.counter.Store(currentVal)
 
 	newResetAfter := tn + tick.Nanoseconds()
 	if !c.resetAt.CAS(resetAfter, newResetAfter) {
 		// We raced with another goroutine trying to reset, and it also reset
 		// the counter to 1, so we need to reincrement the counter.
-		return c.counter.Inc()
+		return c.counter.Inc(), 0
 	}
+	return
```

Comment:
Naked returns are generally frowned on, prefer the explicit `return currentVal, preResetVal`

> Naked returns are okay if the function is a handful of lines. Once it's a medium sized function, be explicit with your return values. 

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters